src/service-workers/offline-first.js

```diff
@@ -0,0 +1,348 @@
+            clearContentCache();
+            break;
+        case "revision-check":
+            this.checkRevision(e.data.url);
+            break;
+        case "pjax":
+            this.pjax(e.data.url, e.data.requestId, e.data.currentUrl, e.data.followRedirects);
+            break;
+        case "prefetch":
+            const existingQueue = this.prefetch.length;
+            this.prefetchQueue = [...this.prefetchQueue, ...e.data.urls];
+            if (!existingQueue) {
+                this.prefetch();
+            }
+            break;
+        default:
+            console.error(`Unknown Service Worker message type: ${type}`);
+            break;
+    }
+});
+
+self.addEventListener("fetch", event => {
+    const noCache = event.request.url.match(new RegExp(REPLACE_WITH_NO_CACHE_PATTERN));
+    if (noCache || event.request.method !== "GET") {
+        // Use a passthrough to ignore the caching system
+        event.respondWith(fetch(event.request));
+    } else {
+        const isResource = event.request.url.match(/(\.js)$|(\.css)$|(\.mjs)$|(\.cjs)$|(\.png)$|(\.jpg)$|(\.gif)$|(\.webp)$|(\.jpeg)$/gi);
+        const cacheName = isResource ? resourcesCacheId : contentCacheId;
+
+        event.respondWith(
+            caches.match(event.request).then(response => {
+                if (response) {
+                    return response;
+                }
+
+                return fetch(event.request, {
+                    redirect: "follow",
+                    credentials: "include",
+                }).then(response => {
+                    if (!response || response.status !== 200 || response.type !== "basic" || response.headers.get("PWA-Cache") === "no-cache" || response.redirected) {
+                        return response;
+                    }
+
+                    var responseToCache = response.clone();
+
+                    caches.open(cacheName).then(cache => {
+                        cache.put(event.request, responseToCache);
+                    });
+                    return response;
+                });
+            })
+        );
+    }
+});
+
+function clearContentCache() {
+    caches.keys().then(cacheNames => {
+        return Promise.all(
+            cacheNames.map(cacheName => {
+                if (cacheName.match("content")) {
+                    return caches.delete(cacheName);
+                }
+            })
+        );
+    });
+}
+
+function informClientOfCachebustValues(maximumContentPrompts, contentCacheDuration, url) {
+    clients = self.clients.matchAll().then(clients => {
+        clients.map(client => {
+            if (client.visibilityState === "visible" && client.url === url) {
+                client.postMessage({
+                    type: "cachebust",
+                    max: parseInt(maximumContentPrompts),
+                    contentCacheExpires: parseInt(contentCacheDuration),
+                });
+            }
+        });
+    });
+}
+
+async function cachebust(url) {
```

Comment:
Couple of things:

1. We need to be better with our variable names. Let's be more descriptive with what we are doing with the variables.

1. Per our conversation, this code is slow, we are doing a sequential fetch requests. Let's change to call both, and handle the response underneath.